core/src/main/java/overflowdb/storage/NodeDeserializer.java

```diff
@@ -88,7 +91,8 @@ public final NodeRef deserializeRef(byte[] bytes) throws IOException {
     Object[] res = new Object[propertyCount * 2];
     int resIdx = 0;
     for (int propertyIdx = 0; propertyIdx < propertyCount; propertyIdx++) {
-      final String key = intern(unpacker.unpackString());
+      int keyId = unpacker.unpackInt();
+      final String key = intern(storage.reverseLookupStringToIntMapping(keyId));
```

Comment:
I think that `storage.reverseLookupStringToIntMapping(keyId)` should already intern -- no need to look up in the hashmap all the time.

Reply:
I agree, but there's a few other issues with the interner, so I'll bring them in via another PR to not mix things too much.